src/Groups/directproducts.jl

```diff
@@ -0,0 +1,486 @@
+################################################################################
+#
+#  Managing direct products
+#
+################################################################################
+
+export
+    as_matrix_group,
+    as_perm_group,
+    as_polycyclic_group,
+    cartesian_power,
+    direct_product,
+    embedding,
+    factorofdirectproduct,
+    homomorphism_of_semidirect_product,
+    isfull_direct_product,
+    isfull_semidirect_product,
+    isfull_wreath_product,
+    number_of_factors,
+    projection,
+    semidirect_product,
+    sub,
+    wreath_product,
+    write_as_full
+
+
+"""
+    direct_product(L::AbstractVector{<:GAPGroup})
+    direct_product(L::GAPGroup...)
+Return the direct product of the groups in the vector `L`.
```

Comment:
But in the second case, L isn't a vector, it's a tuple. So perhaps like this:
```suggestion
Return the direct product of the groups in the collection `L`.
```
(And then this immediately suggests one could implement this more generally by supporting any iterator / iterable as argument ;-) but let's not get overboard for now)